### PR TITLE
 Add ODS for aten.nll_loss2d Op

### PR DESCRIPTION
**Pull request by rahuls-cerebras**
This PR
- Adds aten.nll_loss2d_forward and aten.nll_loss2d_backward op
- Also add UTs

### CODE REVIEW

File: python/torch_mlir_e2e_test/test_suite/nll_loss.py [hunk 1]
```diff
@@ -133,6 +133,52 @@ def forward(self, x, y):
 def NllLossModule_ignore_index_out_of_bounds_basic(module, tu: TestUtils):
   module.forward(tu.rand(2, 3), tu.randint(2, low=0, high=3))
 
+class NllLossModule_2D(torch.nn.Module):
```
Comment (ramiro050) [inline python/torch_mlir_e2e_test/test_suite/nll_loss.py:136]: If you just need the ODS, there's no need to add tests. Tests should be added in the same PR as the op lowering so that it is easier for reviewers to see the different edge cases being tested.

**Review by ramiro050 — CHANGES_REQUESTED** [review]

File: python/torch_mlir_e2e_test/test_suite/nll_loss.py (hunk 1 above)
Comment (rahuls-cerebras) [reply python/torch_mlir_e2e_test/test_suite/nll_loss.py]: fixed

**Comment by rahuls-cerebras** [comment]
The error doesnt look like related to my changes

**Comment by rahuls-cerebras** [comment]
build-test

**Comment by rahuls-cerebras** [comment]
@ramiro050 can you please review

**Review by ramiro050 — APPROVED** [review]
